### PR TITLE
Created FuncSegment and RepFuncSegment backbone models in BackboneModels.js

### PR DESCRIPTION
**Pull request by i-m-foster**
we would appreciate feedback ty :)

### CODE REVIEW

File: leaf-ui/rappid-extensions/BackboneModels.js [hunk 1]
```diff
@@ -1 +1,27 @@
-/** This file contains backbone model representations of the original model objects - WIP */
\ No newline at end of file
+/** This file contains backbone model representations of the original model objects - WIP */
+
+
+// FuncSegmentModel from FuncSegment class in modelObjects.js
+var FuncSegmentModel = Backbone.Model.extend({
+    idAttribute: "uid",
+
+    // initialize function - could also save all of the input parameters into one variable
```
Comment (leshghi) [inline leaf-ui/rappid-extensions/BackboneModels.js]: So you actually have to save all the input parameters into one option variable I am pretty sure. You can test this by trying to create a FuncSegmentModel instance and then printing it out to see how/if the parameters were assigned properly

**Review by leshghi — COMMENTED** [review]
Look into the options param I mentioned - otherwise looks good from a technical standpoint. I assigned @amgrubb to take a look as well for structural things.

File: leaf-ui/rappid-extensions/BackboneModels.js [hunk 2]
```diff
@@ -1 +1,29 @@
...
+
+
+// FuncSegmentModel from FuncSegment class in modelObjects.js
+var FuncSegmentModel = Backbone.Model.extend({
+    idAttribute: "uid",
+
+    // initialize function - could also save all of the input parameters into one variable 
+    // 'options' and then use that like Yesuegen 
+    initialize: function (options) {
+        this.funcType = options.funcType;
```
Comment (amgrubb) [inline leaf-ui/rappid-extensions/BackboneModels.js]: Can we change `funcX` to `refEvaluationValue`.
We are going to take func out of each of these names.
`type`, `refEvaluationValue`, `startTP`, `stopTP`.

File: leaf-ui/rappid-extensions/BackboneModels.js [hunk 3]
```diff
@@ -1 +1,29 @@
-/** This file contains backbone model representations of the original model objects - WIP */
\ No newline at end of file
+/** This file contains backbone model representations of the original model objects - WIP */
+
+
+// FuncSegmentModel from FuncSegment class in modelObjects.js
+var FuncSegmentModel = Backbone.Model.extend({
```
Comment (amgrubb) [inline leaf-ui/rappid-extensions/BackboneModels.js]: `FunctionSegmentBBM`

File: leaf-ui/rappid-extensions/BackboneModels.js [hunk 4]
```diff
@@ -1 +1,29 @@
...
+});
+
+//RepFuncSegmentMOdel from RepFuncSegment class in modelObjects.js
+var RepFuncSegmentModel = Backbone.Model.extend({
+    idAttribute: "uid",
+
+    initialize: function(options) { 
+        this.functionSegList = options.functionSegList; 
+        this.repNum = options.repNum; 
+        this.absTime = options.absTime;
```
Comment (amgrubb) [inline leaf-ui/rappid-extensions/BackboneModels.js]: `this.absTime` to `this.absRepTime`

**Review by amgrubb — CHANGES_REQUESTED** [review]
Just some renaming.

File: leaf-ui/rappid-extensions/BackboneModels.js [hunk 5]
```diff
@@ -1 +1,29 @@
...
+    initialize: function (options) {
+        this.type = options.type;
+        this.refEvaluationValue = options.refEvaluationValue;
+        this.startTP = options.startTP;
+        this.stopTP = options.stopTP;
+    },
+});
+
+//RepFuncSegmentMOdel from RepFuncSegment class in modelObjects.js
+var RepFuncSegmentModel = Backbone.Model.extend({
```
Comment (leshghi) [inline leaf-ui/rappid-extensions/BackboneModels.js]: @amgrubb Should this be RepFuncSegmentBBM?

Comment (amgrubb) [reply leaf-ui/rappid-extensions/BackboneModels.js]: Yup or RepFunctionSegmentBBM

**Review by amgrubb — APPROVED** [review]

**Pull request merged**